src/modlunky2/ui/trackers/common.py

```diff
@@ -54,29 +59,51 @@ def send(self, command: CommonCommand, data):
     def die(self, message):
         self.send(CommonCommand.DIE, message)
 
-    def _run(self):
-        shutting_down = False
+    def wait(self):
+        self.proc = None
+        self.send(CommonCommand.WAIT, None)
 
+    def _attach(self):
         pid = find_spelunky2_pid()
         if pid is None:
-            self.die("Failed to find running Spel2.exe")
-            return
+            # This is fine, we'll try again later
+            return False
 
-        self.proc = Spel2Process.from_pid(pid)
-        if self.proc is None:
+        proc = Spel2Process.from_pid(pid)
+        if proc is None:
             self.die("Failed to open handle to Spel2.exe")
-            return
+            return False
 
-        if self.proc.state is None:
+        if proc.get_feedcode() is None:
+            # Game might still be starting, we should try again
+            return False
+
+        if proc.state is None:
             self.die("Failed to open handle to expected array of bytes")
-            return
+            return False
+
+        self.proc = proc
+        self.initialize()
+        return True
+
+    def _run(self):
+        shutting_down = False
+        if not self._attach():
+            self.wait()
 
         while True:
             if not shutting_down and self.shut_down:
                 shutting_down = True
                 break
 
-            self._really_poll()
+            if self.proc is None:
+                self._attach()
+            elif self.proc.running():
+                self._really_poll()
+            else:
+                self.wait()
+                self._attach()
+
             time.sleep(0.1)
```

Comment:
One concern i have here is that the interval for polling for memory and the interval for polling for the exe are tied together. We might want to have a separate interval if it's running vs attaching.

Reply:
what i'm thinking is maybe something like class variables

POLL_INTERVAL = 0.1
ATTACH_INTERVAL = 1

then if `self._attach()` is False with use the ATTACH_INTERVAL instead of the poll interval

Reply:
Done, assuming I understood correctly... my Python is so rusty.